topic/autoscaling/autoscale.py

```diff
@@ -0,0 +1,93 @@
+import requests
+import time
+from datetime import datetime
+from requests.auth import HTTPBasicAuth
+
+# Authentication for API requests
+auth_data = HTTPBasicAuth('YOUR API KEY', 'YOUR API SECRET')
+
+# Organization and cluster IDs (to be filled by the user)
+organization_id = 'FILL IN YOUR ORGANIZATION ID'
+cluster_id = 'FILL IN YOUR CLUSTER ID'
+
+# Date format for parsing datetime strings
+date_format = '%Y-%m-%dT%H:%M:%S.%f'
+
+# Function to fetch metrics from CrateDB Cloud for a specific organization
+def get_metrics(organization_id):
+    api_url = f'https://console.cratedb.cloud/api/v2/organizations/{organization_id}/metrics/prometheus/'
+    headers = {'accept': 'plain/text'}
+    response = requests.get(api_url, auth=auth_data)
+    return response.text
+
+# Function to calculate the average number of shards per node for a given cluster
+def num_shards(metrics, cluster_id):
+    headers = {'accept': 'plain/text'}
+    api_url = f"https://console.cratedb.cloud/api/v2/organizations/{organization_id}/metrics/prometheus/"
+    response = requests.get(api_url, auth=auth_data, headers=headers)
+    if response.status_code == 200:
+        lines = response.text.split('\n')
+        total_shard_count = 0
+        shard_count_instances = 0
+        for line in lines:
+            if "crate_node" in line and "shard_stats" in line and cluster_id in line and "property=\"total\"" in line:
+                shard_count = float(line.split()[1])
+                total_shard_count += shard_count
+                shard_count_instances += 1
+        if shard_count_instances == 0:
+            return None  # Return None if no shard counts were found
+        else:
+            return total_shard_count / shard_count_instances  # Calculate and return the average
+    else:
+        print(f"Failed to retrieve metrics. Status code: {response.status_code}")
```

Comment:
As far as `print` is concerned in general, using it in a logger-like style is somewhat discouraged. For such a small example script it's not a big deal, but on the other hand, using a logger is also straightforward. In the sense of developing a habit of using loggers, I argue for using them:

```python
import logging

logging.getLogger().setLevel('INFO')

logging.info('INFO message')
logging.debug('DEBUG message')
```

Reply:
Changed that.